### PR TITLE
datacards for WptoTpb

### PR DESCRIPTION
**Pull request by jsiado**
Datacards for W' -> T'b in the semileptonic channel for 2017 data

### CODE REVIEW

File: bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/WpToTpBottom/WpToTpBottom_conf/Wb_madspin_card.dat [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+#*    The MadGraph5_aMC@NLO Development Team - Find us at   *
+#*    https://server06.fynu.ucl.ac.be/projects/madgraph     *
+#*                                                          *
+#************************************************************
+
+set ms_dir ./madspingrid
+set max_running_process 1
+
+set Nevents_for_max_weigth 250 # number of events for the estimate of the max. weight
+#set BW_cut 15                # cut on how far the particle can be off-shell
```
Comment (efeyazgan) [inline bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/WpToTpBottom/WpToTpBottom_conf/Wb_madspin_card.dat]: please remove this unused line

File: bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/WpToTpBottom/WpToTpBottom_conf/Wb_madspin_card.dat [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+set max_weight_ps_point 400  # number of PS to estimate the maximum for each event
+
+# specify the decay for the final state particles
```
Comment (efeyazgan) [inline bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/WpToTpBottom/WpToTpBottom_conf/Wb_madspin_card.dat]: please remove this unused line

File: bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/WpToTpBottom/WpToTpBottom_conf/Wb_madspin_card.dat [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+decay t > w+ b, w+ > all all
+decay t~ > w- b~, w- > all all
+decay w+ > all all
+decay w- > all all
+
+# running the actual code
```
Comment (efeyazgan) [inline bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/WpToTpBottom/WpToTpBottom_conf/Wb_madspin_card.dat]: please remove this unused line

File: bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/WpToTpBottom/WpToTpBottom_conf/Wb_proc_card.dat [hunk 4]
```diff
@@ -0,0 +1,16 @@
+set group_subprocesses Auto
+set ignore_six_quark_processes False
+set loop_optimized_output True
+set complex_mass_scheme False
+import model sm
+define p = g u c d s u~ c~ d~ s~
+define j = g u c d s u~ c~ d~ s~
+define l+ = e+ mu+
```
Comment (efeyazgan) [inline bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/WpToTpBottom/WpToTpBottom_conf/Wb_proc_card.dat]: Leptons and neutrinos are defined but not used so may be removed

File: bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/WpToTpBottom/WpTpb_Wp1500Nar_Tp700NarRH_Wb/WpTpb_Wp1500Nar_Tp700NarRH_Wb_run_card.dat [hunk 5]
```diff
@@ -0,0 +1,282 @@
...
+     6500     = ebeam2  ! beam 2 total energy in GeV
+#*********************************************************************
+# Beam polarization from -100 (left-handed) to 100 (right-handed)    *
+#*********************************************************************
+        0     = polbeam1 ! beam polarization for beam 1
+        0     = polbeam2 ! beam polarization for beam 2
+#*********************************************************************
+# PDF CHOICE: this automatically fixes also alpha_s and its evol.    *
+#*********************************************************************
+# 'cteq6l1'    = pdlabel     ! PDF set (lhapdf for using lhapdf)
```
Comment (efeyazgan) [inline bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/WpToTpBottom/WpTpb_Wp1500Nar_Tp700NarRH_Wb/WpTpb_Wp1500Nar_Tp700NarRH_Wb_run_card.dat]: please remove this line

File: bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/WpToTpBottom/WpTpb_Wp1500Nar_Tp700NarRH_Wb/WpTpb_Wp1500Nar_Tp700NarRH_Wb_run_card.dat [hunk 6]
```diff
@@ -0,0 +1,282 @@
...
+#
+#*********************************************************************
+# Store info for systematics studies                                 *
+# WARNING: If use_syst is T, matched Pythia output is                *
+#          meaningful ONLY if plotted taking matchscale              *
+#          reweighting into account!                                 *
+#*********************************************************************
+   T  = use_syst      ! Enable systematics studies
+#
+#**************************************
```
Comment (efeyazgan) [inline bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/WpToTpBottom/WpTpb_Wp1500Nar_Tp700NarRH_Wb/WpTpb_Wp1500Nar_Tp700NarRH_Wb_run_card.dat]: please remove this and the commented lines below - if these exist in other runcards, please do the same.

**Comment by efeyazgan** [comment]
I left individual comments to some lines. 

In addition to that, there are too many files. Instead of this, could you please put representative cards only and if you have a script to generate all other cards from the representative ones?

**Comment by jsiado** [comment]
@efeyazgan You said "remove this line" but there are several lines in the comment. Did you mean to remove a particular line or all the lines in the comment. For example for the last comment Should I remove all lines from 269 to 272.
There are files for each mass point. That was what I see in other cards I was using to create these.

File: bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/WpToTpBottom/WpTpb_Wp1500Nar_Tp700NarRH_Wb/WpTpb_Wp1500Nar_Tp700NarRH_Wb_run_card.dat (hunk 6 above)
Comment (efeyazgan) [reply bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/WpToTpBottom/WpTpb_Wp1500Nar_Tp700NarRH_Wb/WpTpb_Wp1500Nar_Tp700NarRH_Wb_run_card.dat]: Please ignore this comment.

**Comment by efeyazgan** [comment]
"remove all lines from 269 to 272" --> was a mistake. Sorry, ignore that. Others are relevant. 

Please just keep one representative mass point. If we have too many files, github will slow down.

**Comment by jsiado** [comment]
Do I ignore all the comments? Or the last one? If I left only one representative mass point, How do I tell you what mass points we want?

**Comment by efeyazgan** [comment]
Just the last one. 

The genproductions area is to keep the cards and settings. Which mass points you want will be defined in the corresponding requests.

File: bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/WpToTpBottom/WpTpb_Wp1500Nar_Tp700NarRH_Wb/WpTpb_Wp1500Nar_Tp700NarRH_Wb_run_card.dat (hunk 6 above)
Comment (efeyazgan) [reply bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/WpToTpBottom/WpTpb_Wp1500Nar_Tp700NarRH_Wb/WpTpb_Wp1500Nar_Tp700NarRH_Wb_run_card.dat]: (As I wrote before) Please ignore this particular comment and do the other comments. Thanks.

**Comment by jsiado** [comment]
@efeyazgan I think I have done all the changes now. Let me know if I missed something

File: bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/WpToTpBottom/WpToTpBottom_conf/Wb_proc_card.dat [hunk 7]
```diff
@@ -0,0 +1,20 @@
+set group_subprocesses Auto
+set ignore_six_quark_processes False
+set loop_optimized_output True
+set complex_mass_scheme False
+
+import model sm
+
+define p = g u c d s u~ c~ d~ s~
```
Comment (perrozzi) [inline bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/WpToTpBottom/WpToTpBottom_conf/Wb_proc_card.dat]: Hi, you should remove the duplicate import as only the last one is valid.
also, we recommend to remove the proton and jet definitions
https://twiki.cern.ch/twiki/bin/viewauth/CMS/Moriond18MC#Checks_to_the_datacards_BEFORE_T

Comment (jsiado) [reply bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/WpToTpBottom/WpToTpBottom_conf/Wb_proc_card.dat]: But there are two different model. Should I delete all lepton, proton, and  jet definition?

**Comment by jsiado** [comment]
@perrozzi Does the merged mean I can go ahead and generate the gridpacks?

**Comment by perrozzi** [comment]
yes I would say so

**Pull request merged**